### PR TITLE
[Jest] Migrate `vector_tile_worker_source.test.js`

### PR DESCRIPTION
**Pull request by wipfli**
- Move and rename
- Run jscodeshift
- Run custom codemods and lint
- Manual edits
- Wrap in describe

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Manually test the debug page.
 - [ ] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog".
 - [ ] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

File: src/source/vector_tile_worker_source.test.ts [hunk 1]
```diff
@@ -260,12 +262,12 @@ describe('VectorTileWorkerSource provides resource timing information (fallback
     const sampleMarks = [100, 350];
     const marks = {};
     const measures = {};
-    t.stub(perf, 'getEntriesByName').callsFake((name) => { return measures[name] || []; });
-    t.stub(perf, 'mark').callsFake((name) => {
+    window.performance.getEntriesByName = jest.fn().mockImplementation(name => (measures[name] || []));
```
Comment (HarelM) [inline src/source/vector_tile_worker_source.test.ts]: We need to make sure somehow this does not affect other tests as this is not "restored" at the end of this test.
I would consider moving this to the before each and restore it in after each...?

Comment (wipfli) [reply src/source/vector_tile_worker_source.test.ts]: https://github.com/maplibre/maplibre-gl-js/pull/745/commits/95331c184257b1afc466acc3c675725ba4bc2c91 clears mocks after every test. Before each test, we call `setPerformance()` which initializes the performance mocks.

Comment (wipfli) [reply src/source/vector_tile_worker_source.test.ts]: Though, `mockClear` is maybe not what we want, as it does not remove implementations: https://jestjs.io/docs/mock-function-api#mockfnmockclear

Comment (wipfli) [reply src/source/vector_tile_worker_source.test.ts]: Should I use `jest.resetAllMocks()` after each test instead?

Comment (HarelM) [reply src/source/vector_tile_worker_source.test.ts]: You need to store the original implementation in the before and restore it in the after, I think...

Comment (wipfli) [reply src/source/vector_tile_worker_source.test.ts]: Are you referring to the original implementation of `window.performance.getEntriesByName` or also other functions?

Comment (HarelM) [reply src/source/vector_tile_worker_source.test.ts]: Everything that starts with `window`

Comment (wipfli) [reply src/source/vector_tile_worker_source.test.ts]: I'm a bit lost here. Can you make a code suggestion?

Comment (HarelM) [reply src/source/vector_tile_worker_source.test.ts]: Something like:
```js
let originalFn;
beforeEach(
    ...
    originalFn = window.performance.getEntriesByName;
    window.performance.getEntriesByName = jest.fn...
}
afterEach(
    ...
   window.performance.getEntriesByName = originalFn;
```

Comment (HarelM) [reply src/source/vector_tile_worker_source.test.ts]: The same is needed for `mark` and `measure` I believe.
Other than that this is approved.

**Comment by birkskyum** [comment]
@wipfli , is this ready for review?

**Review by birkskyum — APPROVED** [review]

**Review by HarelM — APPROVED** [review]
Yes, great!

**Comment by wipfli** [comment]
Thanks both!

**Pull request merged**